controller/src/main/java/org/jboss/as/controller/capability/RuntimeCapability.java

```diff
@@ -391,9 +462,7 @@ private Builder(String baseName, boolean dynamic, T runtimeAPI) {
          * @param requirements the capability names
          * @return the builder
          *
-         * @deprecated nothing is currently done with this data and unless a true use is implemented this method may be removed
          */
-        @Deprecated
```

Comment:
Use case is still here, it is now actually in use. 
See https://github.com/wildfly/wildfly-core/pull/2103/files#diff-340f1dd448e96a0a6a57e4028fb436faR243
We talked about deferring the actual validation to validateCapabilites() in CapabilityRegistry

Recorder now uses this information to know if requirement should be added or not.